akka-actor/src/main/scala/akka/actor/dungeon/DeathWatch.scala

```diff
@@ -11,7 +11,7 @@ import akka.event.AddressTerminatedTopic
 
 private[akka] trait DeathWatch { this: ActorCell ⇒
 
-  private var watching: Set[ActorRef] = ActorCell.emptyActorRefSet
+  private var watching: Map[ActorRef, Option[Any]] = Map.empty
```

Comment:
Ok, that wouldn't grow it though, just add a level of indirection with another object than watching. Maybe just document that care must be taken accessing it, if hiding is not an option.